### ISSUE TITLE
@argument completion with Typescript AddOn in Visual Studio Code

### ISSUE THREAD

**Issue by csprocket777** [lifeart/els-addon-docs#19]
I have a component I'm developing leveraging ember-cli-typescript and ember-cli-addon-docs using the YUIDoc generator.

I'm following the YUIDoc spec to document my arguments in my interface like this:
```typescript
interface MyComponentArgs extends BaseComponentArgs {
    /**
     * A Custom CSS class to be added to the base component classes.
     * @argument class
     * @type {String}
     * @public
     */
    class: string;
}
```
(open to syntax corrections if there are any)

## My issue:
When in Visual Studio Code, I can type `<` and see the component name autocompletion, but when I go to then type `@` to get access to the autocomplete menu for the arguments of the component, none of the arguments appear.

**Example:** *(while in VSCode)*
```handlebars
<MyComponent @[expecting auto complete with at least "@class" here..]
```

I'm struggling to determine where the issue is. Any help would be appreciated.

**Comment by lifeart** [comment]
Hi @csprocket777! Thank you for question!

els-addon-docs use fetch to load API version for addon see https://github.com/lifeart/els-addon-docs/blob/master/lib/doc-resolver.js#L20

do you have generated json in your system? or it's public addon?

Also, make sure you have https://marketplace.visualstudio.com/items?itemName=lifeart.vscode-ember-unstable installed

**Comment by lifeart** [comment]
Looks like best option for you is - https://github.com/typed-ember/glint

**Comment by csprocket777** [comment]
I'm developing my addon locally right now. Once complete, it'll be in a private repo and the docs will be behind an SSO endpoint. 
Do the docs need to be bundled with it?

**Comment by lifeart** [comment]
If we talking about `els-addon-docs` usage, we need to figure out how to get generated doc's 'json' of locally developed addon. Once it's clear - it's minore change to add support.

**Comment by csprocket777** [comment]
Would that be something as simple as a package.json entry for the addon itself? Assuming we're talking about ensuring Visual Studio code's plugin system being able to consume that value for each addon installed, right?

**Comment by lifeart** [comment]
@csprocket777 correct, right now this addon rely on publically available json (by http), but this logic could be extended to look into `node_modules/addon-name/...some-path.json` or `dist/some-path.json` to resolve it

**Comment by csprocket777** [comment]
I'm confused, isn't the docs json in `dist/docs/[addonName]/[addonName].json`? Can't it be read from there? Is that something that just needs to be baked into the `package.json` in some property?

**Comment by lifeart** [comment]
It's not for addons, downloaded using npm/yarn, but if it's true for locally stored addon, yes, it's could be used for documentation`

>> Is that something that just needs to be baked into the package.json in some property?
- likely not, we should check for `dist/docs/[addonName]/[addonName].json` file on FS.

**Comment by csprocket777** [comment]
What can be done to help make this real?

**Comment by lifeart** [comment]
@csprocket777 it would be great to have reproduction repo to test this addon on

**Comment by csprocket777** [comment]
I'm happy to be the guinea pig, but as my challenge stems from my addon being a non-public one and being behind our corporate firewall, I'm not sure how to provide you a repo to test with.

Although, it could be something as simple as producing an addon that is hosted by a private github/gitlab repo so that the docs aren't publicly available but only when it's installed locally... Thoughts?

**Comment by lifeart** [comment]
yes, simple minimal reproduction is fine

**Comment by csprocket777** [comment]
Alright, well I can't figure out why my doc site isn't working but here it is while in public form.
https://github.com/csprocket777/private-addon-doc-test

Let me know if this will work as a test case and we can try making the docs private to do the test with.

**Comment by lifeart** [comment]
Looks like there is no docs without build is created

**Comment by lifeart** [comment]
think we could do here - run new process, to get json output from it :/

**Comment by csprocket777** [comment]
Is there something else you need me to do on this? Happy to help.

**Comment by lifeart** [comment]
@csprocket777 we need to find a way to run "compilation" task from https://github.com/ember-learn/ember-cli-addon-docs on addon itself (programmatically)

**Comment by csprocket777** [comment]
So then we can't do anything without the ember-learn team? How do we bring attention to this?

**Comment by lifeart** [comment]
@csprocket777 we could try to figure out this logic manually

**Comment by csprocket777** [comment]
How does it know where online to look for the documentation json?

**Comment by lifeart** [comment]
@csprocket777 sample - https://miguelcobain.github.io/ember-yeti-table/

**Comment by csprocket777** [comment]
How does the els-addon-docs addon asses the addons installed in a project, in order to know where to look for a particular addon's docs?

**Comment by lifeart** [comment]
it's look into addon's package json to find doc website to load docs https://github.com/lifeart/els-addon-docs/blob/master/lib/doc-resolver.js#L46
